docs/azure/sdk/thread-safety.md

```diff
@@ -0,0 +1,121 @@
+---
+title: Thread safety with the Azure SDK for .NET
+description: Learn about thread safety in Azure SDK client objects and how this design impacts client lifetime management.
+ms.date: 05/24/2021
+ms.author: pakrym
+author: pakrym
+---
+
+# Thread safety and client lifetime management for Azure SDK objects
+
+This article helps you understand thread safety issues when using the Azure SDK. It also discusses how the design of the SDK impacts client lifetime management. You'll learn why it's unnecessary to dispose of Azure SDK client objects.
+
+## Thread safety
+
+All Azure SDK client objects are thread-safe and independent of each other. This design ensures that reusing client instances is always safe, even across threads. For example, the following code launches multiple tasks but is thread safe:
+
+```csharp
+var client = new SecretClient(new Uri("<secrets_endpoint>"), new DefaultAzureCredential());
```

Comment:
Wrap this line to prevent horizontal scrolling in docs.

```suggestion
var client = new SecretClient(
    new Uri("<secrets_endpoint>"), new DefaultAzureCredential());
```